src/operator/quantization/mkldnn/mkldnn_quantized_batch_norm.cc

```diff
@@ -0,0 +1,129 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file mkldnn_quantized_batch_norm.cc
+ * \brief
+ * \author Yixin Bao
+*/
+
+#if MXNET_USE_MKLDNN == 1
+#include "../../nn/mkldnn/mkldnn_batch_norm-inl.h"
+#include "../quantization_utils.h"
+
+namespace mxnet {
+namespace op {
+
+static void MKLDNNQuantizedBatchNormForward(const nnvm::NodeAttrs &attrs, const OpContext &ctx,
+                                            const std::vector<NDArray> &in_data,
+                                            const std::vector<OpReqType> &req,
+                                            const std::vector<NDArray> &outputs) {
+  CHECK_EQ(in_data.size(), 7U);
+  CHECK_EQ(outputs.size(), 3U);
+
+  TmpMemMgr::Get()->Init(ctx.requested[batchnorm::kTempSpace]);
+  const BatchNormParam &param = nnvm::get<BatchNormParam>(attrs.parsed);
+  const NDArray &data = in_data[quantized_batchnorm::kData];
+  const size_t channelAxis = static_cast<size_t>(
+      param.axis < 0 ? static_cast<int>(data.shape().ndim()) + param.axis : param.axis);
+  const int channel_count = data.shape()[channelAxis];
+  const float min_data = in_data[quantized_batchnorm::kDataMin].data().dptr<float>()[0];
+  const float max_data = in_data[quantized_batchnorm::kDataMax].data().dptr<float>()[0];
+  const float max_abs_data = std::max(std::abs(min_data), std::abs(max_data));
+
+  float *min_output_ptr = outputs[quantized_batchnorm::kOutMin].data().dptr<float>();
+  float *max_output_ptr = outputs[quantized_batchnorm::kOutMax].data().dptr<float>();
+  if (param.min_calib_range.has_value() && param.max_calib_range.has_value()) {
+    *max_output_ptr =
+        std::max(std::abs(param.min_calib_range.value()), std::abs(param.max_calib_range.value()));
+    *min_output_ptr = -*max_output_ptr;
+  } else {
+    LOG(FATAL) << "min_calib_range or max_calib_range is not available. Quantized BN currently "
+                  "don't support calib_mode=None";
+  }
+
+  unsigned flags = mkldnn::use_global_stats | mkldnn::use_scale_shift;
+  auto &fwd = GetBNForward<float>(param, ctx, data, flags);
+  const mkldnn::memory &weight_mem = fwd.GetWeight();
+  CHECK_EQ(weight_mem.get_primitive_desc().get_size(), channel_count * sizeof(float) * 2);
+  float *weight_buf = reinterpret_cast<float *>(weight_mem.get_data_handle());
+
+  NDArray gamma = in_data[quantized_batchnorm::kGamma];
+  NDArray beta = in_data[quantized_batchnorm::kBeta];
+  float *gamma_ptr = gamma.data().dptr<float>();
```

Comment:
Seems gamma only used here. Perhaps you can remove it as 
`float *gamma_ptr = in_data[quantized_batchnorm::kGamma].data().dptr<float>();`